### PR TITLE
Add ctest und unittest infrastructure

### PR DESCRIPTION
**Pull request by vvolkl**
Already shows first issue: #138

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+add_subdirectory(unittest)
+
+add_test(NAME examples/flavor/generic-analysis
```
Comment (clementhelsens) [inline tests/CMakeLists.txt:4]: I have removed this one, please use ```examples/FCCee/higgs/mH-recoil/mumu``` instead

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+        # TODO: check if possible to make eos world readable
+        #COMMAND python examples/FCCee/flavour/generic-analysis/analysis.py root://eospublic.cern.ch//eos/experiment/fcc/ee/generation/DelphesEvents/spring2021/IDEA/p8_ee_Zbb_ecm91_EvtGen_Bc2TauNuTAUHADNU/events_131527278.root
+        COMMAND python examples/FCCee/flavour/generic-analysis/analysis.py https://fccsw.web.cern.ch/testsamples/fccanalyses/eos/experiment/fcc/ee/generation/DelphesEvents/spring2021/IDEA/p8_ee_Zbb_ecm91_EvtGen_Bc2TauNuTAUHADNU/events_131527278.root
```
Comment (clementhelsens) [inline tests/CMakeLists.txt:7]: same comment as above

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+
+add_subdirectory(unittest)
+
+add_test(NAME examples/FCCee/higgs/mH-recoil/mumu
+        # TODO: check if possible to make eos world readable
+        COMMAND python config/FCCAnalysisRun.py examples/FCCee/higgs/mH-recoil/mumu/analysis_stage1.py --output myoutput.root --files-list root://eospublic.cern.ch//eos/experiment/fcc/ee/generation/DelphesEvents/spring2021/IDEA/p8_ee_Zbb_ecm91_EvtGen_Bc2TauNuTAUHADNU/events_131527278.root
+        #COMMAND python config/FCCAnalysisRun.py examples/FCCee/higgs/mH-recoil/mumu/analysis_stage1.py --output myoutput.root --files-list https://fccsw.web.cern.ch/testsamples/fccanalyses/eos/experiment/fcc/ee/generation/DelphesEvents/spring2021/IDEA/p8_ee_Zbb_ecm91_EvtGen_Bc2TauNuTAUHADNU/events_131527278.root
```
Comment (clementhelsens) [inline tests/CMakeLists.txt:7]: is there an issue getting the file from the web?

Comment (kjvbrt) [reply tests/CMakeLists.txt]: When the file is retrieved from fccsw.web.cern.ch using the https method `TWebFile` complains with error 503. Getting the file from other webserver through https works without issue.

**Pull request merged**